pykafka/balancedconsumer.py

```diff
@@ -547,6 +561,10 @@ def _rebalance(self):
                         raise
                     log.info('Unable to acquire partition %s. Retrying', ex.partition)
                     time.sleep(i * (self._rebalance_backoff_ms / 1000))
+            if self._partitions_assigned_callback is not None:
+                self._partitions_assigned_callback(self, new_partitions)
```

Comment:
If the user wants to use this callback to load offsets from their custom store via `reset_offsets()`, we would need some way to prevent `BalancedConsumer.consume()` from serving messages between L553 and here, otherwise it would in the mean time return messages at whatever offset we pulled from kafka.